### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by justinboswell**
Note that MQTT needs a docs pass, and I need to add docs once HttpConnection is converted to using events.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lib/common/http.ts [hunk 1]
```diff
@@ -17,16 +17,24 @@ import { InputStreamBase } from "./io";
...
 
+    /** Construct from a collection of [name, value] pairs */
     constructor(headers: HttpHeader[] = []) {
         for (const header of headers) {
             this.add(header[0], header[1]);
         }
     }
 
+    /** Add a name/value pair 
+     * @param name - The header name
```
Comment (graebm) [inline lib/common/http.ts]: trivial: I mildly prefer:
```
/**
 * First line
 * next line
 */
```
over
```
/** First line
 * next line
 */
```

but I'm also whatever

File: lib/native/io.ts [hunk 2]
```diff
@@ -15,32 +15,60 @@
 
 import crt_native = require('./binding');
 import { NativeResource } from "./native_resource";
-import { ResourceSafe } from '../common/resource_safety';
-import { InputStreamBase } from '../common/io';
+import { InputStreamBase, TlsVersion } from '../common/io';
 import { Readable } from 'stream';
+export { TlsVersion } from '../common/io';
```
Comment (ColdenCullen) [inline lib/native/io.ts]: double import?

Comment (ColdenCullen) [reply lib/native/io.ts]: JK fuck exports

Comment (justinboswell) [reply lib/native/io.ts]: Nope, import and export are separate/different.

**Pull request merged**